source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRContentStackView.mm

```diff
@@ -56,25 +59,47 @@ - (ContainerStyle)getStyle
     return _style;
 }
 
-- (void)setBackgroundColor:(ContainerStyle)style
-                hostConfig:(std::shared_ptr<HostConfig> const &)config
++ (UIColor *)colorFromString:(const std::string&)colorString
 {
-    _style = style;
-    long num = 0;
-    if(style == ContainerStyle::Emphasis)
-    {
-        num = std::stoul(config->containerStyles.emphasisPalette.backgroundColor.substr(1), nullptr, 16);
-    }
-    else
-    {
-        num = std::stoul(config->containerStyles.defaultPalette.backgroundColor.substr(1), nullptr, 16);
-    }
+    long num = std::stoul(colorString.substr(1), nullptr, 16);
+
+    return [UIColor colorWithRed:((num & 0x00FF0000) >> 16) / 255.0
+                    green:((num & 0x0000FF00) >>  8) / 255.0
+                     blue:((num & 0x000000FF)) / 255.0
+                    alpha:((num & 0xFF000000) >> 24) / 255.0];
+}
+
++ (CGFloat)borderWidthFromString:(const std::string&)thicknessString
+{
+    return std::stof(thicknessString);
+}
+
+- (ContainerStyleDefinition&)paletteForHostConfig:(std::shared_ptr<HostConfig> const &)config
+{
+    return (_style == ContainerStyle::Emphasis)
+        ? config->containerStyles.emphasisPalette
+        : config->containerStyles.defaultPalette;
+}
+
+- (void)setBackgroundColorWithHostConfig:(std::shared_ptr<HostConfig> const &)config
+{
+    UIColor *color = [[self class] colorFromString:[self paletteForHostConfig:config].backgroundColor];
```

Comment:
what is advantage of using [self class] vs ACRContentStackView? 
if compiler optimizes the code, wouldn't it replace the above call with ACRContentStackView?

Reply:
Advantage is that you don't have to spell out the exact class name ;)

Reply:
o.k. i was concerned more about having an additional call. but i guess it's o.k. since I assume it will be optimized by compiler.

---
In reply to: [169714151](https://github.com/Microsoft/AdaptiveCards/pull/1150#discussion_r169714151) [](ancestors = 169714151)